NonSucking.Framework.Serialization/GeneratedSerializerCode.cs

```diff
@@ -0,0 +1,151 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using VaVare.Statements;
+
+namespace NonSucking.Framework.Serialization;
+
+public class GeneratedSerializerCode
+{
+    public List<SerializerVariable> VariableDeclarations { get; } = new();
+    public List<StatementSyntax> Statements { get; } = new();
+
+    public IEnumerable<StatementSyntax> ToMergedBlock()
+    {
+        foreach (var declaration in VariableDeclarations)
+        {
+            yield return declaration.ToDeclarationAndAssignment();
+        }
+
+        foreach (var statement in Statements)
+        {
+            yield return statement;
+        }
+    }
+
+    public IEnumerable<StatementSyntax> MergeBlocksSeperated(GeneratedSerializerCode other,
+        Func<SerializerVariable, LocalDeclarationStatementSyntax> variableTransformer = null)
+    {
+        foreach (var declaration in VariableDeclarations)
+        {
+            var newDeclaration = variableTransformer == null
```

Comment:
Why don't you use is null?

```suggestion
            var newDeclaration = variableTransformer is null
```